### PR TITLE
Shell interpreter test and doc

### PR DESCRIPTION
**Pull request by fvaleri**
### What is this PR for?
Small refactoring to Shell interpreter.
Add missing unit test and initial doc.

### What type of PR is it?
Improvement, Documentation, Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/ZEPPELIN/
* Put link here, and add [ZEPPELIN-*Jira number*] in PR title, eg. [ZEPPELIN-533]

### How should this be tested?
Outline the steps to test the PR here.

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by jongyoul** [comment]
Could you please rebase this from current master?

**Comment by fvaleri** [comment]
@jongyoul done and CI is ok.

File: shell/src/test/java/org/apache/zeppelin/shell/ShellInterpreterTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    shell.open();
+    InterpreterContext context = new InterpreterContext("", "1", "", "", null, null, null, null, null, null, null);
+    InterpreterResult result = new InterpreterResult(Code.ERROR);
+    if (System.getProperty("os.name").startsWith("Windows")) {
+      result = shell.interpret("dir", context);
+    } else {
+      result = shell.interpret("ls", context);
+    }
+    // System.out.println(result.message());
+    assertEquals(InterpreterResult.Code.SUCCESS, result.code());
```
Comment (bzz) [inline shell/src/test/java/org/apache/zeppelin/shell/ShellInterpreterTest.java]: It's better not to leave comments hanging and remove un-used ones.

**Comment by bzz** [comment]
Great clean up and improvement @fvaleri!

While you and @jongyoul are both here - how do you think guys, would it make sense to take care of Shell part of the [ZEPPELIN-804](https://issues.apache.org/jira/browse/ZEPPELIN-804) as prart of this PR as well?

File: shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java [hunk 2]
```diff
@@ -43,24 +46,31 @@
...
+
+  static {
+    Interpreter.register(
+      "sh", "sh",
+      ShellInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+        .add(SHELL_COMMAND_TIMEOUT, DEFAULT_COMMAND_TIMEOUT,
+          "Shell command time out in millisecs. Default = " + DEFAULT_COMMAND_TIMEOUT).build()
+    );
+  }
```
Comment (bzz) [inline shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java]: Is there a reason to use static interpreter registration mechanism here, aftert [ZEPPELIN-923](https://issues.apache.org/jira/browse/ZEPPELIN-923) AKA #956 ?

Comment (fvaleri) [inline shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java]: thanks @bzz for pointing out, I will take care of this ASAP

**Comment by fvaleri** [comment]
@bzz @jongyoul Now should be ok.

**Comment by jongyoul** [comment]
The failure is flaky and irrelevant but we need to ensure all other tests passed. Could you please re-trigger CI?

**Comment by fvaleri** [comment]
Another CI job failed even this time on zeppelin-web module.
One is still running but it is very slow (32 min now).
```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.25:bower (bower install) on project zeppelin-web: Failed to run task: 'bower --allow-root install' failed. (error code 1) -> [Help 1]
```
Anyway, [this one](https://travis-ci.org/apache/zeppelin/builds/141304231) passed and it's the same code, except for a couple on unused imports that I removed.

**Comment by jongyoul** [comment]
@fvaleri Got it. LGTM.

**Pull request closed without merging**